services/autoscaling/tests/unit/test_core_settings.py

```diff
@@ -197,11 +198,32 @@ def test_EC2_INSTANCES_ALLOWED_TYPES_passing_valid_image_tags(  # noqa: N802
 def test_EC2_INSTANCES_ALLOWED_TYPES_empty_not_allowed(  # noqa: N802
     app_environment: EnvVarsDict, monkeypatch: pytest.MonkeyPatch
 ):
+    assert app_environment["AUTOSCALING_EC2_INSTANCES"] == "{}"
     monkeypatch.setenv("EC2_INSTANCES_ALLOWED_TYPES", "{}")
 
-    with pytest.raises(ValidationError):
+    # test child settings
+    with pytest.raises(ValidationError) as err_info:
+        EC2InstancesSettings.create_from_envs()
+    assert err_info.value.errors()[0]["loc"] == ("EC2_INSTANCES_ALLOWED_TYPES",)
+
+    # now as part of AUTOSCALING_EC2_INSTANCES: EC2InstancesSettings | None
+    assert os.environ["AUTOSCALING_EC2_INSTANCES"] == "{}"
+
+    with pytest.raises(ValidationError) as err_info:
+        ApplicationSettings.create_from_envs(AUTOSCALING_EC2_INSTANCES={})
+
+    before = err_info.value.errors()
+
+    with pytest.raises(ValidationError) as err_info:
         ApplicationSettings.create_from_envs()
 
+    assert err_info.value.errors() == before
+
+    # removing any value for AUTOSCALING_EC2_INSTANCES
+    monkeypatch.delenv("AUTOSCALING_EC2_INSTANCES")
+    settings = ApplicationSettings.create_from_envs()
+    assert settings.AUTOSCALING_EC2_INSTANCES is None
+
```

Comment:
this gives the feeling of many tests in a test that says something else.

Reply:
true. I split them